### ISSUE TITLE
Investigate if we also need aria-label when we already have a label

### ISSUE THREAD

**Issue by thrbnhrtmnn** [deven-org/boiler#528]
#### Description / User Story

When checking the props documentation the question arose, if i.e. a textButton with an icon always needs an aria-label and a label prop, or if the aria-label is not needed or even not wanted, when the label is already there.

With this task we want to do some investigation and figure out, what the WCAG 2.1 criteria says. In the end it should be clear how to handle label and aria-label not only for a textButton, but also for all other components that have a label and possibly also an aria-label.

----
#### Acceptance Criteria
- [ ] A clear definition has been found, how to handle aria-label and label and if they both are needed or only one.

----
#### Background information
- This task is an requirement for #475 
- We thought also about including a logic to use the value of the "label" prop in the "aria-label" prop in case both are needed for a component
